### PR TITLE
fix: improve git remoteUrl parsing

### PR DESCRIPTION
**Pull request by ndom91**
## ☕️ Reasoning

- We weren't parsing some different types/variants of git remote URLs
- https://github.com/gitbutlerapp/gitbutler/issues/4132

## 🧢 Changes

- New dependency `git-url-parse` (https://github.com/IonicaBizau/git-url-parse) [`~3.5kb minified + gzipped`]
	- GitHub only supports a subset of the remote URL types that `git` supports, so maybe the manual parsing we had was sufficient for the time being
	- But having less error-prone / battle tested parsing will make it easier for us to support other git services in the future and avoid other edge-cases
- Add a few initial vitest tests for the `GitHubService` to ensure that `owner`/`repo` are parsed out as expected

Out of the 22 URLs I added to test, the previous implementation only parsed 13/22 correctly.

### CODE REVIEW

File: app/src/lib/github/service.test.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
+import { GitHubService } from './service';
+import { BehaviorSubject } from 'rxjs';
+import { expect, test, describe } from 'vitest';
+
+const exampleRemoteUrls = [
+	'ssh://user@host.xz:123/org/repo.git/',
```
Comment (Byron) [inline app/src/lib/github/service.test.ts:6]: I am loving this!

**Review by Byron — COMMENTED** [review]
That's a great call, and of course the JS world has a parser already which probably covers 99% of the common cases as well by now.

So I'd say, despite my recommendation, let's use it until it breaks :).

**Comment by Caleb-T-Owens** [comment]
Love this @ndom91!! 

If you're up for it, do you want to change `convertRemoteToWebUrl` to something like: `GitUrlParse("...").toString("https").replace(/.git$/, '')`?

**Review by Caleb-T-Owens — APPROVED** [review]
